### ISSUE TITLE
探索部分中间计算使用 BF16 提升 TPU 吞吐

### ISSUE THREAD

**Issue by sii-xinglong** [sgl-project/sglang-jax#829]
## 问题描述

当前所有中间计算均强制转换为 float32（`gla_pallas.py` 多处 `.astype(jnp.float32)`）。在 TPU v4/v5 上，BF16 matmul 吞吐量约为 FP32 的 2 倍（MXU BF16 peak ≈ 2× FP32 peak）。部分精度不敏感的中间计算可以使用 BF16 提升吞吐量。

## 当前实现

```python
# _matmul_f32, L88-92 — 所有 matmul 统一走 FP32
def _matmul_f32(a, b, *, precision=None):
    a_f32 = a.astype(jnp.float32)
    b_f32 = b.astype(jnp.float32)
    ...
    return jnp.matmul(a_f32, b_f32, precision=precision)

# _compute_chunk_forward_with_A, L200-204 — 所有输入统一转 FP32
q_f32 = q.astype(jnp.float32)
k_f32 = k.astype(jnp.float32)
v_f32 = v.astype(jnp.float32)
g_cumsum_f32 = g_cumsum.astype(jnp.float32)
state_f32 = state.astype(jnp.float32)
```

## 精度分析

各 matmul 的精度敏感度：

| 计算 | 精度敏感度 | BF16 可行性 | 说明 |
| -- | -- | -- | -- |
| `A·V` (o_intra) | **低** | 可行 | A 已经是 masked & scaled，V 值域有限 |
| `qg·state` (o_inter) | 中 | 需验证 | state 可能 accumulate 较大值 |
| `Q·K^T` (A matrix) | **高** | 不推荐 | 涉及 exp 稳定化，精度敏感 |
| `k_discount^T·V` (delta) | 中 | 需验证 | state update 精度影响长期记忆 |
| Backward 各 matmul | **高** | 不推荐 | 梯度精度直接影响训练稳定性 |

### 阶段 1：保守优化（低风险）

对精度不敏感的 `A·V` intra-chunk matmul 使用 BF16：

```python
# o_intra: A 已经是 lower-triangular masked, 值域受控
o_intra = jnp.matmul(a.astype(jnp.bfloat16), v.astype(jnp.bfloat16)).astype(jnp.float32)
```

### 阶段 2：可配置精度

提供环境变量或参数控制不同 matmul 的精度：

```python
LYNX_GLA_INTRA_PRECISION = os.environ.get("LYNX_GLA_INTRA_PRECISION", "bf16")
LYNX_GLA_INTER_PRECISION = os.environ.get("LYNX_GLA_INTER_PRECISION", "f32")
```

### 验证方法

* 使用现有 `test_chunk_gla_pallas.py` 的测试用例，对比 full-FP32 vs mixed-precision 的输出误差
* 关注指标：cosine similarity ≥ 0.9999, atol ≤ 1e-3
* 在长序列（T=8192+）上验证 state 数值稳定性

## 预期收益

* `A·V` matmul 使用 BF16：吞吐提升约 1.5-2x（该 matmul 占 forward 计算量的 \~25%）
* 整体 forward pass 预计提速 10-20%（仅对保守方案）

## 风险

* BF16 动态范围有限（\~1e-38 到 \~3.4e38），需确保中间值不会 overflow
* 训练 loss 可能出现微小波动，需端到端验证
* 需要充分测试确保梯度精度不受影响

## 相关代码

* `gla_pallas.py` L81-92: `_matmul_f32`（统一 FP32 matmul）
* `gla_pallas.py` L40-57: `_default_f32_matmul_precision`（精度配置）
* `gla_pallas.py` L200-212: forward 中间计算
* `tests/test_chunk_gla_pallas.py`: 精度验证测试
